py-utils/src/utils/msg_bus/message_bus/bus/bus_frame.py

```diff
@@ -0,0 +1,142 @@
+import logging
+log = logging.getLogger(__name__)
+
+# Kafka-Python
+from message_bus.adaptee import KafkaAdaptee
+from message_bus.bus.callback import MyCallback
+from message_bus.template import kafkaFactory, ConfluentFactory
+from message_bus.template import Factory, Singleton
+from message_bus.bus.topic_schema import TopicSchema
+from message_bus.bus.topic import Topic
+from message_bus.utils import log_decorator
+
+
+class Bus(metaclass=Singleton):
+
+    @log_decorator
+    def __init__(self, mq_bus_name, bus_callback=MyCallback()):
+        # log.debug("Init bus class")
+        self.config = None
+        self.notifier = None #notifier callable
+        self.mapper = {}
+        self.callables = {}
+        self.mq_bus_name = mq_bus_name
+        self.bus_callback = bus_callback
+        self.m_factory = Factory({
+            "kafka-python": kafkaFactory,
+            "confluent-kafka": ConfluentFactory
+        })
+        self.__load_adaptor(mq_bus_name)
+        if self.config is not None:
+            self.__load_topic()
+
+        self.schema = TopicSchema()
+        self.client_list = []
+
+    def __load_adaptor(self, mq_bus_name):
```

Comment:
Please rename all the adaptor to adapter. The meaning are different. In our context we need adapter, I think.

Reply:
Renamed